### ISSUE TITLE
[Watch/Future] keep Manifest v2

### ISSUE THREAD

**Issue by tomByrer** [Alex313031/thorium#79]
Google's in development Manifest v3 will remove the ability to redirect/block URLs:

https://www.youtube.com/watch?v=8KWCLhHrblE&list=PLakykuPxo3chMNq-SOFIkkh3QgwSqXV0n&index=1&t=331s

**Comment by Alex313031** [comment]
@tomByrer Yeah I know. Im a user of ublock origin. Theres alot of other extensions I also use that probably wont get updated. Its a shitty move by google. I will be adding the enterprise policy patch to keep MV2 for as long as possible, but unless another chromium fork comes up with a patch, I dont think I would be able to create one myself. In that case, I might just use the last version of thorium that supports it for myself, and suggest other users do so if they really need a mv2 extension. Otherwise I dont really know what to say.

**Comment by RobloxEnderman** [comment]
You could add an ad blocker built into the browser (kinda like brave), then just add an option to turn it on on first setup. That will at least help with the ad block issue.

**Comment by brucethemoose** [comment]
So Chromium 112 based builds is when this will become an issue, right?

**Comment by Alex313031** [comment]
@brucethemoose Well that is when it will have it turned off by default, but there will still be enterprise policy for another year to allow manifest v2 extensions. When that comes around I will be forcing the code that checks for that policy to be set to true always, that way manifest v2 will work in thorium all the way until they completely remove the code that actually loads/runs manifest v2 extensions and chrome apps.

**Comment by SViN24** [comment]
@Alex313031 from what I understand Brave uses an integrated adblocker engine https://github.com/brave/adblock-rust

Would you like to integrate it on Thorium?

**Comment by gz83** [comment]
> @Alex313031 from what I understand Brave uses an integrated adblocker engine https://github.com/brave/adblock-rust
> 
> Would you like to integrate it on Thorium?

I recommend using the same type of extension for ad filtering.

@SViN24

**Comment by proofy** [comment]
Because this project is “only” an optimization of Chromium, the developer cannot be required to change the manifest in all future versions of Chromium. In my opinion, that would be too much work.
As you can read in the Chromium blog, Manifest V2 will be supported until at least June 2025. 
https://blog.chromium.org/2024/05/manifest-v2-phase-out-begins.html
However, it will be deactivated.
Here Thorium could leave it always enabled as a default setting?

**Comment by gz83** [comment]
It is currently on version M150, and MV2 support is still being maintained via patches.
